interactive_engine/executor/ir/runtime/src/process/entry.rs

```diff
@@ -0,0 +1,609 @@
+        Some(self)
+    }
+}
+
+#[derive(Debug, Clone, Default, PartialEq, PartialOrd, Eq, Hash)]
+pub struct CollectionEntry {
+    pub inner: Vec<DynEntry>,
+}
+
+impl_as_any!(CollectionEntry);
+
+impl Entry for CollectionEntry {
+    fn get_type(&self) -> EntryDataType {
+        EntryDataType::Collection
+    }
+}
+
+impl Element for CollectionEntry {
+    fn as_graph_element(&self) -> Option<&dyn GraphElement> {
+        None
+    }
+
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+
+    fn as_borrow_object(&self) -> BorrowObject {
+        BorrowObject::None
+    }
+}
+
+impl Encode for CollectionEntry {
+    fn write_to<W: WriteExt>(&self, writer: &mut W) -> std::io::Result<()> {
+        self.inner.write_to(writer)
+    }
+}
+
+impl Decode for CollectionEntry {
+    fn read_from<R: ReadExt>(reader: &mut R) -> std::io::Result<Self> {
+        let inner = <Vec<DynEntry>>::read_from(reader)?;
+        Ok(CollectionEntry { inner })
+    }
+}
+
+impl TryFrom<result_pb::Element> for DynEntry {
+    type Error = ParsePbError;
+    fn try_from(e: result_pb::Element) -> Result<Self, Self::Error> {
+        if let Some(inner) = e.inner {
+            match inner {
+                result_pb::element::Inner::Vertex(v) => {
+                    let vertex = Vertex::try_from(v)?;
+                    Ok(DynEntry::new(vertex))
+                }
+                result_pb::element::Inner::Edge(e) => {
+                    let edge = Edge::try_from(e)?;
+                    Ok(DynEntry::new(edge))
+                }
+                result_pb::element::Inner::GraphPath(p) => {
+                    let path = GraphPath::try_from(p)?;
+                    Ok(DynEntry::new(path))
+                }
+                result_pb::element::Inner::Object(o) => {
+                    let obj = Object::try_from(o)?;
+                    Ok(DynEntry::new(obj))
+                }
+            }
+        } else {
+            Err(ParsePbError::EmptyFieldError("element inner is empty".to_string()))?
+        }
+    }
+}
+
+impl TryFrom<result_pb::Entry> for DynEntry {
+    type Error = ParsePbError;
+
+    fn try_from(entry_pb: result_pb::Entry) -> Result<Self, Self::Error> {
+        if let Some(inner) = entry_pb.inner {
+            match inner {
+                result_pb::entry::Inner::Element(e) => Ok(e.try_into()?),
+                result_pb::entry::Inner::Collection(c) => {
+                    let collection = CollectionEntry {
+                        inner: c
+                            .collection
+                            .into_iter()
+                            .map(|e| e.try_into())
+                            .collect::<Result<Vec<_>, Self::Error>>()?,
+                    };
+                    Ok(DynEntry::new(collection))
+                }
+            }
+        } else {
+            Err(ParsePbError::EmptyFieldError("entry inner is empty".to_string()))?
+        }
+    }
+}
+
+impl From<Vertex> for DynEntry {
```

Comment:
Why not implement 
```
impl<E: Entry + 'static> From<E> for DynEntry { ... }
```

Then the `new()` function in `DynEntry` can be removed.

Reply:
Cannot do this since `DynEntry` is also an `Entry`, thus we got the errror:
```
    | impl<E: Entry + 'static> From<E> for DynEntry {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: conflicting implementation in crate `core`:
            - impl<T> From<T> for T;

```